File: _java-archive/acquisition-service/src/main/java/dev/kritchalach/acquisition/domain/PropertySurvey.java
package dev.kritchalach.acquisition.domain;

import jakarta.persistence.Embeddable;
import java.math.BigDecimal;

/**
 * Value Object: PropertySurvey
 * ข้อมูล survey ของ property ที่ Inventory ส่งมา immutable
 */
@Embeddable
public class PropertySurvey {

    private String address;
    private BigDecimal areaSqm;
    private BigDecimal estimatedValue;
    private String zoneType;

    protected PropertySurvey() {}

    public PropertySurvey(String address, BigDecimal areaSqm,
                          BigDecimal estimatedValue, String zoneType) {
        this.address = address;
        this.areaSqm = areaSqm;
        this.estimatedValue = estimatedValue;
        this.zoneType = zoneType;
    }

    public String getAddress() { return address; }
    public BigDecimal getAreaSqm() { return areaSqm; }
    public BigDecimal getEstimatedValue() { return estimatedValue; }
    public String getZoneType() { return zoneType; }
}

File: acquisition-service/src/domain/PropertySurvey.ts
/**
 * Value Object: PropertySurvey
 * ข้อมูล survey ที่ Inventory ส่งมา immutable
 */
export class PropertySurvey {
  constructor(
    public readonly address: string,
    public readonly areaSqm: number,
    public readonly estimatedValue: number,
    public readonly zoneType: string,
  ) {}
}
